### ISSUE TITLE
Axum 0.6.0

### ISSUE THREAD

**Issue by ieow** [oscartbeaumont/rspc#89]
got this issue when using rspc with Axum 0.6.0
```
mismatched types
expected struct `MethodRouter<(), _>`
   found struct `axum::routing::method_routing::MethodRouter`
perhaps two different versions of crate `axum` are being used?
```

**Comment by oscartbeaumont** [comment]
Thanks for reporting. Currently, rspc doesn't support Axum 0.6. I have started work on supporting Axum 0.6 but it's proving challenging due to the way extractors have changed upstream. I am closing in favor of issue #104 which has a more detailed rundown.

**Comment by stewartadam** [comment]
@oscartbeaumont was this resolved for v0.1.3? Release notes say Axum 0.6 support, but I'm still bumping against this.
